src/Cake.Core/Tooling/ToolSettings.cs

```diff
@@ -12,5 +13,11 @@ public class ToolSettings
         /// </summary>
         /// <value>The tool path.</value>
         public FilePath ToolPath { get; set; }
+
+        /// <summary>
+        /// Gets or sets the arguments.
+        /// </summary>
+        /// <value>The arguments.</value>
+        public Action<ProcessArgumentBuilder> ArgumentCustomization { get; set; }
```

Comment:
I kinda disagree, but might be personal preference.
Personally don't think this is unclean, unclear or less fluent.
```csharp
Settings.ArgumentCustomization = builder=>"arg"
```
vs.
```csharp
Settings.ArgumentCustomization = builder => {  
    builder.Clear();
    builder.Append("Arg");
};
```
As the all the extension methods return a ProcessArgumentBuilder it would just work without any code change.

I would just find a `Func` more flexible with little added complexity, would support any scenario the Action does plus a few more.